apps/evans/components/core/AudioPlayer.tsx

```diff
@@ -20,6 +20,27 @@ const AudioPlayer: React.FC<Props> = ({ tracks }) => {
   const [audio, setAudio] = useState<HTMLAudioElement | null>(null);
   const track = tracks[currentTrackIndex];
 
+  useEffect(() => {
+    const trackStorageItem = localStorage.getItem(window.location.pathname);
+    if (trackStorageItem) {
+      let {
+        storedTime,
+        storedTrackIndex,
+      }: { storedTime: number; storedTrackIndex: number } = JSON.parse(trackStorageItem);
```

Comment:
i'd like you to extract a function that takes the location path as an argument and returns `{storedTime: number; storedTrackIndex: number; } | null`

in it, i want you to be a little more defensive. wrap the `JSON.parse()` in a `try/catch` (it can throw), and if you get a value out of parsing it, test that it has the two keys you expect, and that the values for each are `number`s. if it throws, or you encounter anything unexpected, the function would return `null`